File: managed/src/SwiftlyS2.Generated/Natives/EntitySystem.cs
#pragma warning disable CS0649

using System.Buffers;
using System.Text;

namespace SwiftlyS2.Core.Natives;

internal static class NativeEntitySystem {
  private unsafe static delegate* unmanaged<void*, void*, void> _Spawn;
  public unsafe static void Spawn(void* entity, void* keyvalues) {
    _Spawn(entity, keyvalues);
  }
  private unsafe static delegate* unmanaged<void*, void> _Despawn;
  public unsafe static void Despawn(void* entity) {
    _Despawn(entity);
  }
  private unsafe static delegate* unmanaged<byte*, void*> _CreateEntityByName;
  public unsafe static void* CreateEntityByName(string name) {
    var pool = ArrayPool<byte>.Shared;
    var nameLength = Encoding.UTF8.GetByteCount(name);
    var nameBuffer = pool.Rent(nameLength + 1);
    Encoding.UTF8.GetBytes(name, nameBuffer);
    nameBuffer[nameLength] = 0;
    fixed (byte* nameBufferPtr = nameBuffer) {
    var ret = _CreateEntityByName(nameBufferPtr);
    pool.Return(nameBuffer);

    return ret;
  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, int, int, void> _AcceptInputInt32;
  public unsafe static void AcceptInputInt32(void* entity, string input, void* activator, void* caller, int value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputInt32(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, uint, int, void> _AcceptInputUInt32;
  public unsafe static void AcceptInputUInt32(void* entity, string input, void* activator, void* caller, uint value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputUInt32(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, long, int, void> _AcceptInputInt64;
  public unsafe static void AcceptInputInt64(void* entity, string input, void* activator, void* caller, long value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputInt64(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, ulong, int, void> _AcceptInputUInt64;
  public unsafe static void AcceptInputUInt64(void* entity, string input, void* activator, void* caller, ulong value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputUInt64(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, float, int, void> _AcceptInputFloat;
  public unsafe static void AcceptInputFloat(void* entity, string input, void* activator, void* caller, float value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputFloat(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, double, int, void> _AcceptInputDouble;
  public unsafe static void AcceptInputDouble(void* entity, string input, void* activator, void* caller, double value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputDouble(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, bool, int, void> _AcceptInputBool;
  public unsafe static void AcceptInputBool(void* entity, string input, void* activator, void* caller, bool value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AcceptInputBool(entity, inputBufferPtr, activator, caller, value, outputID);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, byte*, int, void> _AcceptInputString;
  public unsafe static void AcceptInputString(void* entity, string input, void* activator, void* caller, string value, int outputID) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {

    var valueLength = Encoding.UTF8.GetByteCount(value);
    var valueBuffer = pool.Rent(valueLength + 1);
    Encoding.UTF8.GetBytes(value, valueBuffer);
    valueBuffer[valueLength] = 0;
    fixed (byte* valueBufferPtr = valueBuffer) {
    _AcceptInputString(entity, inputBufferPtr, activator, caller, valueBufferPtr, outputID);
    pool.Return(inputBuffer);

    pool.Return(valueBuffer);

  }
  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, int, float, void> _AddEntityIOEventInt32;
  public unsafe static void AddEntityIOEventInt32(void* entity, string input, void* activator, void* caller, int value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventInt32(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, uint, float, void> _AddEntityIOEventUInt32;
  public unsafe static void AddEntityIOEventUInt32(void* entity, string input, void* activator, void* caller, uint value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventUInt32(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, long, float, void> _AddEntityIOEventInt64;
  public unsafe static void AddEntityIOEventInt64(void* entity, string input, void* activator, void* caller, long value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventInt64(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, ulong, float, void> _AddEntityIOEventUInt64;
  public unsafe static void AddEntityIOEventUInt64(void* entity, string input, void* activator, void* caller, ulong value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventUInt64(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, float, float, void> _AddEntityIOEventFloat;
  public unsafe static void AddEntityIOEventFloat(void* entity, string input, void* activator, void* caller, float value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventFloat(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, double, float, void> _AddEntityIOEventDouble;
  public unsafe static void AddEntityIOEventDouble(void* entity, string input, void* activator, void* caller, double value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventDouble(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, bool, float, void> _AddEntityIOEventBool;
  public unsafe static void AddEntityIOEventBool(void* entity, string input, void* activator, void* caller, bool value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {
    _AddEntityIOEventBool(entity, inputBufferPtr, activator, caller, value, delay);
    pool.Return(inputBuffer);

  }
  }
  private unsafe static delegate* unmanaged<void*, byte*, void*, void*, byte*, float, void> _AddEntityIOEventString;
  public unsafe static void AddEntityIOEventString(void* entity, string input, void* activator, void* caller, string value, float delay) {
    var pool = ArrayPool<byte>.Shared;
    var inputLength = Encoding.UTF8.GetByteCount(input);
    var inputBuffer = pool.Rent(inputLength + 1);
    Encoding.UTF8.GetBytes(input, inputBuffer);
    inputBuffer[inputLength] = 0;
    fixed (byte* inputBufferPtr = inputBuffer) {

    var valueLength = Encoding.UTF8.GetByteCount(value);
    var valueBuffer = pool.Rent(valueLength + 1);
    Encoding.UTF8.GetBytes(value, valueBuffer);
    valueBuffer[valueLength] = 0;
    fixed (byte* valueBufferPtr = valueBuffer) {
    _AddEntityIOEventString(entity, inputBufferPtr, activator, caller, valueBufferPtr, delay);
    pool.Return(inputBuffer);

    pool.Return(valueBuffer);

  }
  }
  }
  private unsafe static delegate* unmanaged<void*, bool> _IsValidEntity;
  public unsafe static bool IsValidEntity(void* entity) {
    var ret = _IsValidEntity(entity);
    return ret;
  }
  private unsafe static delegate* unmanaged<void*> _GetGameRules;
  public unsafe static void* GetGameRules() {
    var ret = _GetGameRules();
    return ret;
  }
  private unsafe static delegate* unmanaged<void*> _GetEntitySystem;
  public unsafe static void* GetEntitySystem() {
    var ret = _GetEntitySystem();
    return ret;
  }
}

File: src/scripting/engine/entitysystem.cpp
/************************************************************************************************
 * SwiftlyS2 is a scripting framework for Source2-based games.
 * Copyright (C) 2025 Swiftly Solution SRL via Sava Andrei-Sebastian and it's contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 ************************************************************************************************/

#include <api/interfaces/manager.h>
#include <scripting/scripting.h>

#include <entityhandle.h>

void Bridge_EntitySystem_Spawn(void* pEntity, void* pKeyValues)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->Spawn(pEntity, pKeyValues);
}

void Bridge_EntitySystem_Despawn(void* pEntity)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->Despawn(pEntity);
}

void* Bridge_EntitySystem_CreateEntityByName(const char* name)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    return entsystem->CreateEntityByName(name);
}

void Bridge_EntitySystem_AcceptInputInt32(void* pEntity, const char* input, void* pActivator, void* pCaller, int32_t value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputUInt32(void* pEntity, const char* input, void* pActivator, void* pCaller, uint32_t value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputInt64(void* pEntity, const char* input, void* pActivator, void* pCaller, int64_t value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputUInt64(void* pEntity, const char* input, void* pActivator, void* pCaller, uint64_t value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputFloat(void* pEntity, const char* input, void* pActivator, void* pCaller, float value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputDouble(void* pEntity, const char* input, void* pActivator, void* pCaller, double value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputBool(void* pEntity, const char* input, void* pActivator, void* pCaller, bool value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AcceptInputString(void* pEntity, const char* input, void* pActivator, void* pCaller, const char* value, int32_t outputID)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AcceptInput(pEntity, input, pActivator, pCaller, value, outputID);
}

void Bridge_EntitySystem_AddEntityIOEventInt32(void* pEntity, const char* input, void* pActivator, void* pCaller, int32_t value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventUInt32(void* pEntity, const char* input, void* pActivator, void* pCaller, uint32_t value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventInt64(void* pEntity, const char* input, void* pActivator, void* pCaller, int64_t value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventUInt64(void* pEntity, const char* input, void* pActivator, void* pCaller, uint64_t value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventFloat(void* pEntity, const char* input, void* pActivator, void* pCaller, float value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventDouble(void* pEntity, const char* input, void* pActivator, void* pCaller, double value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventBool(void* pEntity, const char* input, void* pActivator, void* pCaller, bool value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

void Bridge_EntitySystem_AddEntityIOEventString(void* pEntity, const char* input, void* pActivator, void* pCaller, const char* value, float delay)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    entsystem->AddEntityIOEvent(pEntity, input, pActivator, pCaller, value, delay);
}

bool Bridge_EntitySystem_IsValidEntity(void* pEntity)
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    return entsystem->IsValidEntity(pEntity);
}

void* Bridge_EntitySystem_GetGameRules()
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    return entsystem->GetGameRules();
}

void* Bridge_EntitySystem_GetEntitySystem()
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    return entsystem->GetEntitySystem();
}


void* Bridge_EntitySystem_GetFirstActiveEntity()
{
    auto entsystem = g_ifaceService.FetchInterface<IEntitySystem>(ENTITYSYSTEM_INTERFACE_VERSION);
    return entsystem->GetEntitySystem()->m_EntityList.m_pFirstActiveEntity;
}


bool Bridge_EntitySystem_EntityHandleIsValid(uint32 ihandle)
{
    CEntityHandle handle(ihandle);
    return handle.IsValid() && handle.Get() != nullptr;
}

void* Bridge_EntitySystem_EntityHandleGet(uint32 ihandle)
{
    CEntityHandle handle(ihandle);
    return handle.Get();
}

int Bridge_EntitySystem_GetEntityHandleFromEntity(CEntityInstance* pEntity)
{
    return pEntity->GetRefEHandle().ToInt();
}

DEFINE_NATIVE("EntitySystem.Spawn", Bridge_EntitySystem_Spawn);
DEFINE_NATIVE("EntitySystem.Despawn", Bridge_EntitySystem_Despawn);
DEFINE_NATIVE("EntitySystem.CreateEntityByName", Bridge_EntitySystem_CreateEntityByName);
DEFINE_NATIVE("EntitySystem.AcceptInputInt32", Bridge_EntitySystem_AcceptInputInt32);
DEFINE_NATIVE("EntitySystem.AcceptInputUInt32", Bridge_EntitySystem_AcceptInputUInt32);
DEFINE_NATIVE("EntitySystem.AcceptInputInt64", Bridge_EntitySystem_AcceptInputInt64);
DEFINE_NATIVE("EntitySystem.AcceptInputUInt64", Bridge_EntitySystem_AcceptInputUInt64);
DEFINE_NATIVE("EntitySystem.AcceptInputFloat", Bridge_EntitySystem_AcceptInputFloat);
DEFINE_NATIVE("EntitySystem.AcceptInputDouble", Bridge_EntitySystem_AcceptInputDouble);
DEFINE_NATIVE("EntitySystem.AcceptInputBool", Bridge_EntitySystem_AcceptInputBool);
DEFINE_NATIVE("EntitySystem.AcceptInputString", Bridge_EntitySystem_AcceptInputString);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventInt32", Bridge_EntitySystem_AddEntityIOEventInt32);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventUInt32", Bridge_EntitySystem_AddEntityIOEventUInt32);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventInt64", Bridge_EntitySystem_AddEntityIOEventInt64);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventUInt64", Bridge_EntitySystem_AddEntityIOEventUInt64);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventFloat", Bridge_EntitySystem_AddEntityIOEventFloat);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventDouble", Bridge_EntitySystem_AddEntityIOEventDouble);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventBool", Bridge_EntitySystem_AddEntityIOEventBool);
DEFINE_NATIVE("EntitySystem.AddEntityIOEventString", Bridge_EntitySystem_AddEntityIOEventString);
DEFINE_NATIVE("EntitySystem.IsValidEntity", Bridge_EntitySystem_IsValidEntity);
DEFINE_NATIVE("EntitySystem.GetGameRules", Bridge_EntitySystem_GetGameRules);
DEFINE_NATIVE("EntitySystem.GetEntitySystem", Bridge_EntitySystem_GetEntitySystem);
DEFINE_NATIVE("EntitySystem.EntityHandleIsValid", Bridge_EntitySystem_EntityHandleIsValid);
DEFINE_NATIVE("EntitySystem.EntityHandleGet", Bridge_EntitySystem_EntityHandleGet);
DEFINE_NATIVE("EntitySystem.GetEntityHandleFromEntity", Bridge_EntitySystem_GetEntityHandleFromEntity);
DEFINE_NATIVE("EntitySystem.GetFirstActiveEntity", Bridge_EntitySystem_GetFirstActiveEntity);